agent/consul/prepared_query_endpoint.go

```diff
@@ -534,6 +534,11 @@ func (p *PreparedQuery) execute(query *structs.PreparedQuery,
 		nodes = nodeMetaFilter(query.Service.NodeMeta, nodes)
 	}
 
+	// Apply the service metadata filters, if any.
+	if len(query.Service.ServiceMeta) > 0 {
+		nodes = serviceMetaFilter(query.Service.ServiceMeta, nodes)
+	}
+
```

Comment:
Does this mean we are actually add a new feature to prepared queries in general to filter on service meta?

I'm cool with that, but we should consider calling that out in CHANGELOG/docs.

Reply:
Yeah, it is a new feature. Its bringing ServiceMeta up to par with tags for prepared queries so we can try to transition away from tags more.

I updated the docs for ServiceMeta in the information about prepared queries but agree it probably deserves to be called out in the CHANGELOG.